core/src/main/java/io/substrait/expression/proto/ExpressionProtoConverter.java

```diff
@@ -204,6 +204,19 @@ public Expression visit(io.substrait.expression.Expression.ListLiteral expr) {
         });
   }
 
+  @Override
+  public Expression visit(io.substrait.expression.Expression.EmptyListLiteral expr)
+      throws RuntimeException {
+    return lit(
+        builder -> {
+          var protoListType = expr.getType().accept(typeProtoConverter);
+          // For empty lists, the literal's own nullable field is ignored in favor
+          // of the nullability of the enclosed Type.List.
```

Comment:
For safety I'd argue that all writers should update both locations and that the readers should read from the documented location.  A really paranoid reader could check that the two match but I doubt anyone would want to do that.

Reply:
Thanks David, that makes sense.  I've updated this to set the same nullability in both places.